lib/Runtime/Library/JsBuiltIn/JsBuiltIn.js

```diff
@@ -123,4 +124,64 @@
         let o = __chakraLibrary.Object(this);
         return __chakraLibrary.CreateArrayIterator(o, 2 /* ArrayIterationKind.KeyAndValue*/);
     });
-});
\ No newline at end of file
+
+    platform.registerFunction(FunctionsEnum.ArrayIndexOf, function (searchElement, fromIndex) {
+        // ECMAScript 2017 #sec-array.prototype.indexof
+        "use strict";
+
+        if (this === null || this === undefined) {
+            __chakraLibrary.raiseThis_NullOrUndefined("Array.prototype.indexOf");
+        }
+
+        let o;
+        if (__chakraLibrary.isArray(this)) {
+            o = this;
+        }
+        else {
+            o = __chakraLibrary.Object(this);
+        }
+
+        let len = __chakraLibrary.toLength(o["length"]);
```

Comment:
can toLength be defined in the script itself instead of making a call to _chakraLibrary?

Reply:
it probably could be, but it calls toInteger internally so it'd still need a chakraLibrary call unless that was written in JS as well. Might be worth considering for the future as many of the built-ins use that function

Reply:
sound good.

---
In reply to: [186866757](https://github.com/Microsoft/ChakraCore/pull/5102#discussion_r186866757) [](ancestors = 186866757)